### ISSUE TITLE
Feature request: type hints

### ISSUE THREAD

**Issue by dimaqq** [s4v4g3/otel-extensions-python#5]
I'd love to see type hints, so that this library is easier to integrate into existing projects.

**Comment by s4v4g3** [comment]
Thanks for the suggestion.  Can you give me an example of where we're missing type hints?

**Comment by dimaqq** [comment]
Sorry, I've moved on to another project.

**Comment by lenalebt** [comment]
I was having the same request. Basically mypy says:

```
app/accessibility/main.py:33: error: Untyped decorator makes function "assess_accessibility" untyped  [misc]
```

aaaand, well. I'm quite a starter in the python world, coming from strongly typed languages - loosing my types makes me look somewhere else. I don't have that much experience in the python world, so I have no clue how I can work around this right now tbh - except not using this lib for instrumentation :woman_shrugging:

**Comment by s4v4g3** [comment]
I was able to reproduce this by running `mypy` on the example in example.py:

```
$ mypy example

example\example.py:13: error: Untyped decorator makes function "bar" untyped  [misc]
    @instrumented(span_name="my custom span name")
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```     

Working on fixing this in branch `feature/mypy`

**Comment by s4v4g3** [comment]
Resolved in release 1.0.1
